.github/ci/generate-news-feed.js

```diff
@@ -1,79 +1,178 @@
+        id: feedMetadata.link,
+        link: feedMetadata.link,
+        updated: newestPostDate,
+        language: feedItem.language,
+      };
+      const feed = new Feed(feedConfig);
+
+      allPosts.forEach((post) => {
+        const link = feedMetadata["site-root"] + post.path;
+        const feedItemConfig = {
+          title: post.title,
+          id: link,
+          link,
+          content: post.description,
+          date: new Date(post.publishDate * 1000),
+          published: new Date(post.publishDate * 1000),
+        };
+
+        feed.addItem(feedItemConfig);
+      });
+
+      if (!fs.existsSync(feedItem.targetDirectory)) {
+        console.log('Target directory created');
+        fs.mkdirSync(feedItem.targetDirectory);
+      }
+      fs.writeFileSync(targetFile, feed.atom1());
+      console.log('Wrote file to ', targetFile);
+    }
   }
-  fs.writeFileSync(targetFile, feed.atom1());
-  console.log('wrote file to ', targetFile);
 }
 
-async function fetchBlogPosts() {
+/**
+ * Returns a list of properties listed in the block
+ * 
+ * @async
+ * @param {string} endpointUrl blog metadata endpoint
+ * @returns {Array<Post>} the json data object
+*/
+async function fetchBlogPosts(endpointUrl) {
   let offset = 0;
   const allPosts = [];
 
   while (true) {
-    const api = new URL(endpoint);
+    const api = new URL(endpointUrl);
+
     api.searchParams.append('offset', JSON.stringify(offset));
     api.searchParams.append('limit', limit);
-    const response = await fetch(api, {});
-    const result = await response.json();
 
-    allPosts.push(...result.data);
+    const result = await getJsonFromUrl(api);
 
-    if (result.offset + result.limit < result.total) {
-      // there are more pages
-      offset = result.offset + result.limit;
+    if (result) {
+      allPosts.push(...result.data);
+  
+      if (result.offset + result.limit < result.total) {
+        // there are more pages
+        offset = result.offset + result.limit;
+      } else {
+        break;
+      }
     } else {
       break;
     }
   }
   return allPosts;
 }
 
-async function fetchBlogMetadata() {
-  const infoResponse = await fetch(feedInfoEndpoint);
-  const feedInfoResult = await infoResponse.json();
+/**
+ * Returns the Blog Metadata
+ * 
+ * @async
+ * @param {string} endpointUrl blog metadata endpoint
+ * @returns {FeedMetadata} The first feed configuration
+ * @todo This in the future should return all of the configurations for all markets/feeds
+*/
+async function fetchBlogMetadata(endpointUrl) {
+  const feedInfoResult = await getJsonFromUrl(endpointUrl);
+
+  if (!feedInfoResult) {
+    return null;
+  }
+
   return feedInfoResult.data[0];
 }
 
-main()
-  .catch((e) => console.error(e));
+/**
+ * Returns a list of properties listed in the block
+ * @param {string} route get the Json data from the route
+ * @returns {Object} the json data object
+*/
+async function getJsonFromUrl(route) {
```

Comment:
shouldn't this function come from common.js? or is necessary to have it here?

Reply:
This is not part of the FE codebase but a js node script to run on the github ci workflow